third_party/packages/ansi_up/lib/ansi_up.dart

```diff
@@ -2,76 +2,306 @@
+        AnsiUpColor(rgb: [85, 255, 255], class_name: 'ansi-bright-cyan'),
+        AnsiUpColor(rgb: [255, 255, 255], class_name: 'ansi-bright-white'),
+      ]
+    ];
+    palette256 = [];
+    for (var palette in ansiColors) {
+      palette.forEach(palette256.add);
+    }
+    var levels = [0, 95, 135, 175, 215, 255];
+    for (var r = 0; r < 6; ++r) {
+      for (var g = 0; g < 6; ++g) {
+        for (var b = 0; b < 6; ++b) {
+          palette256.add(AnsiUpColor(
+              rgb: [levels[r], levels[g], levels[b]], class_name: 'truecolor'));
+        }
+      }
+    }
+    var greyLevel = 8;
+    for (var i = 0; i < 24; ++i, greyLevel += 10) {
+      palette256.add(AnsiUpColor(
+          rgb: [greyLevel, greyLevel, greyLevel], class_name: 'truecolor'));
+    }
+  }
 
-  external void process_ansi(TextPacket textPacket);
-  external TextWithAttr with_state(TextPacket packet);
+  TextPacket _getNextPacket() {
+    final TextPacket pkt = TextPacket(kind: PacketKind.EOS, text: '', url: '');
+    final int len = _buffer.length;
+    if (len == 0) return pkt;
+    final int pos = _buffer.indexOf('\x1B');
+    if (pos == -1) {
+      pkt.kind = PacketKind.Text;
+      pkt.text = _buffer;
+      _buffer = '';
+      return pkt;
+    }
+    if (pos > 0) {
+      pkt.kind = PacketKind.Text;
+      pkt.text = _buffer.substring(0, pos);
+      _buffer = _buffer.substring(pos);
+      return pkt;
+    }
+    if (pos == 0) {
+      if (len == 1) {
+        pkt.kind = PacketKind.Incomplete;
+        return pkt;
+      }
+      final String nextChar = _buffer[1];
+      if ((nextChar != '[') && (nextChar != ']')) {
+        pkt.kind = PacketKind.ESC;
+        pkt.text = _buffer.substring(0, 1);
+        _buffer = _buffer.substring(1);
+        return pkt;
+      }
+      if (nextChar == '[') {
+        _csiRegex ??= rgx(
+            '\n                        ^                           # beginning of line\n                                                    #\n                                                    # First attempt\n                        (?:                         # legal sequence\n                          \\x1b\\[                      # CSI\n                          ([\\x3c-\\x3f]?)              # private-mode char\n                          ([\\d;]*)                    # any digits or semicolons\n                          ([\\x20-\\x2f]?               # an intermediate modifier\n                          [\\x40-\\x7e])                # the command\n                        )\n                        |                           # alternate (second attempt)\n                        (?:                         # illegal sequence\n                          \\x1b\\[                      # CSI\n                          [\\x20-\\x7e]*                # anything legal\n                          ([\\x00-\\x1f:])              # anything illegal\n                        )\n                    ');
```

Comment:
you can take advantage of multiline dart strings to make this more readable.
```dart
'foo '
'bar '
'baz'
```
is equivalent to 
```dart
''foo bar baz'
```